apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/PipelineManager.java

```diff
@@ -432,28 +485,24 @@ public PipelineJobOutcome call() throws Exception {
 
       try {
         PipelineJobOutcome jobOutcome = wrappedJob.call();
-        jobRecordStore.recordJobCompletion(jobRecord.getId(), jobOutcome);
+        handleJobCompletion(jobRecord.getId(), jobOutcome);
         return jobOutcome;
       } catch (InterruptedException e) {
         /*
          * This indicates that someone has successfully interrupted the job, which should only have
          * happened when we're trying to shut down. Whether or not PipelineJob.isInterruptible() for
          * this job, it's now been stopped, so we should record the cancellation.
          */
-        jobRecordStore.recordJobCancellation(jobRecord.getId());
+        handleJobCancellation(jobRecord.getId());
```

Comment:
I would _like_ remove this entirely, as it's redundant with `PipelineJobCallback.onFailure(Throwable)`, which _should_ always run for cancelations.

However, right now, the code doesn't appear to quite provide the guarantees necessary for that, so let's leave it for now.